pkg/sysregistries/system_registries.go

```diff
@@ -0,0 +1,98 @@
+package sysregistries
+
+import (
+	"github.com/BurntSushi/toml"
+	"io/ioutil"
+)
+
+/*
+systemRegistriesConfPath is the path to the system-wide registry configuration file
+and is used to add/subtract potential registries for obtaining images.
+You can override this at build time with
+-ldflags '-X github.com/containers/image/sysregistries.systemRegistriesConfPath=$your_path'
+*/
+var systemRegistriesConfPath = builtinRegistriesConfPath
+
+/*
+builtinRegistriesConfPath is the path to registry configuration file
+DO NOT change this, instead see systemRegistriesConfPath above.
+*/
+const builtinRegistriesConfPath = "/etc/containers/registries.new"
+
+type tomlConfig struct {
+	Registries registries `toml:"registries"`
+}
+
+type registries struct {
+	SecureRegistries   []string `toml:"secure"`
+	InsecureRegistries []string `toml:"insecure"`
+	BlockRegistries    []string `toml:"block"`
+}
+
+var blankFile = []byte("")
+
+/*
+Reads the global registry file from the filesystem. Returns
+a byte array
+*/
+func readRegistryConf() ([]byte, error) {
+	configBytes, err := ioutil.ReadFile(systemRegistriesConfPath)
+	if err != nil {
+		return blankFile, err
+	}
+	return configBytes, nil
+}
+
+// For mocking in unittests
+var readConf = readRegistryConf
+
+/*
+Loads the registry configuration file from the filesystem and
+then unmarshals it.  Returns the unmarshalled object.
+*/
+func loadRegistryConf() (*tomlConfig, error) {
+	var config tomlConfig
+
+	configBytes, err := readConf()
+	if err != nil {
+		return &config, err
+	}
+
+	err = toml.Unmarshal(configBytes, &config)
+
+	if err != nil {
+		return &config, err
+	}
+
+	return &config, nil
+}
+
+/*
+Returns an array of strings that contain the names
+of the registries as defined in the system-wide
+registries file.  it returns an empty array if none are
+defined
+*/
+func GetRegistries() ([]string, error) {
+	config, err := loadRegistryConf()
+	if err != nil {
+		return nil, err
+	}
+	return config.Registries.SecureRegistries, nil
+}
+
+/*
+Returns an array of strings that contain the names
+of the insecure registries as defined in the system-wide
+registries file.  it returns an empty array if none are
+defined
+*/
+func GetInsecureRegistries() ([]string, error) {
+	config, err := loadRegistryConf()
+	if err != nil {
+		return nil, err
```

Comment:
In idiomatic golang:
if config, err := loadRegistryConf(); err != nil {
    return nil, err
}